app/api/chat/upload/documents/upload.ts

```diff
@@ -0,0 +1,14 @@
+import { loadDocuments, saveDocument } from "./helper";
+import { runPipeline } from "./pipeline";
+
+export async function uploadDocument(
+  raw: string,
+  datasource: string,
+): Promise<string[]> {
+  const [header, content] = raw.split(",");
+  const mimeType = header.replace("data:", "").replace(";base64", "");
+  const fileBuffer = Buffer.from(content, "base64");
+  const documents = await loadDocuments(fileBuffer, mimeType);
+  const { filename } = await saveDocument(fileBuffer, mimeType);
+  return await runPipeline(documents, filename, datasource);
```

Comment:
Change CL to pass `index` to `runPipeline`:
```suggestion
  const index = await getDataSource(datasource, true);
  return await runPipeline(index, documents, filename);
```